### PR TITLE
levelIndexSet() corrected for CpGrid with LGRs

### PR DESCRIPTION
**Pull request by aritorto**
CpGrid::levelIndexSet() is updated to take into account CpGrid with LGRs.

### CODE REVIEW

**Comment by aritorto** [comment]
jenkins build this serial please

File: opm/grid/cpgrid/CpGrid.cpp [hunk 1]
```diff
@@ -921,7 +921,7 @@ const CpGridFamily::Traits::LevelIndexSet& CpGrid::levelIndexSet(int level) cons
 {
     if (level<0 || level>maxLevel())
         DUNE_THROW(GridError, "levelIndexSet of nonexisting level " << level << " requested!");
-    return *current_view_data_->index_set_;
+    return *data_[level] -> index_set_;
```
[the remark this answers is not in the document]
Comment (blattms) [reply opm/grid/cpgrid/CpGrid.cpp:924]: While I reviewed the other one I noticed that we store smart pointers in the container. Hence the condition should probably be `current_view_data_ == data_,back().get()``

[the remark this answers is not in the document]
Comment (aritorto) [reply opm/grid/cpgrid/CpGrid.cpp]: Simplification is done in the last amended commit. Thanks!

**Comment by blattms** [comment]
Thanks a lot. Just running jenkins again.

**Comment by blattms** [comment]
jenkins build this serial please

**Pull request merged**